vault/logical_system_activation_flags_test.go

```diff
@@ -0,0 +1,492 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/enthelpers/activationflags"
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/helper/testhelpers/corehelpers"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/sdk/physical"
+	"github.com/hashicorp/vault/sdk/physical/inmem"
+	"github.com/stretchr/testify/require"
+)
+
+// TestActivationFlags_Read tests the read operation for the activation flags.
+func TestActivationFlags_Read(t *testing.T) {
+	t.Run("given an initial state then read flags and expect all to be unactivated", func(t *testing.T) {
+		core, _, _ := TestCoreUnsealedWithConfig(t, &CoreConfig{})
+
+		resp, err := core.systemBackend.HandleRequest(
+			context.Background(),
+			&logical.Request{
+				Operation: logical.ReadOperation,
+				Path:      prefixActivationFlags,
+				Storage:   core.systemBarrierView,
+			},
+		)
+
+		require.NoError(t, err)
+		require.Equal(t, resp.Data, map[string]interface{}{
+			"activated":   []string{},
+			"unactivated": activationflags.NewFeatureActivationFlags().ValidActivationFeatures(),
```

Comment:
Is it worth enumerating the expected strings here? I can see an argument both ways but as it is this is very close to the actual code being tested so wouldn't catch any bad assumptions.

Since the list is pretty small, it seems OK to hard code it - it will break when we add new ones and need explicit adding but I almost prefer that to make sure the behaviour change is explicitly what the developer adding a new one intends... Not a strong opinion though.